client-sdk/ts-web/core/test/hdkey.test.ts

```diff
@@ -0,0 +1,46 @@
+import {HDKey} from '../src/hdkey';
+import * as adr0008VectorsRaw from './adr-0008-vectors.json';
+
+interface Adr0008Vector {
+    kind: string;
+    bip39_mnemonic: string;
+    bip39_passphrase: string;
+    bip39_seed: string;
+    oasis_accounts: {
+        bip32_path: string;
+        private_key: string;
+        public_key: string;
+        address: string;
+    }[];
+}
+
+const adr0008Vectors: Adr0008Vector[] = adr0008VectorsRaw;
+
+const uint2hex = (array: Uint8Array) => Buffer.from(array).toString('hex');
+
+describe('HDKey', () => {
+    describe('ADR 0008 Vectors', () => {
+        adr0008Vectors.forEach((vector, index) => {
+            it(`Case #${index}`, async () => {
+                // This can be a bit slow on CI servers.
+                jest.setTimeout(10000);
+                const passphrase =
+                    vector.bip39_passphrase && vector.bip39_passphrase !== ''
+                        ? vector.bip39_passphrase
+                        : null;
+
+                for (let account of vector.oasis_accounts) {
+                    const index = Number(account.bip32_path.split('/').pop().replace("'", ''));
```

Comment:
maintainability idea: assert that the beginning of the path starts with the expected values. in case future specs come and someone adds new entries in the json file without knowing that we ignore the beginning